filebeat/input/filestream/input_integration_test.go

```diff
@@ -67,6 +70,50 @@ func TestFilestreamCloseRenamed(t *testing.T) {
 	env.requireOffsetInRegistry(testlogName, len(newerTestlines))
 }
 
+// test_close_removed from test_harvester.py
+func TestFilestreamCloseRemoved(t *testing.T) {
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp := env.mustCreateInput(map[string]interface{}{
+		"paths":                                []string{env.abspath(testlogName) + "*"},
+		"prospector.scanner.check_interval":    "24h",
+		"close.on_state_change.check_interval": "1ms",
+		"close.on_state_change.removed":        "true",
+	})
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	env.startInput(ctx, inp)
+
+	testlines := []byte("first log line\n")
+	env.mustWriteLinesToFile(testlogName, testlines)
+
+	// first event has made it successfully
+	env.waitUntilEventCount(1)
+	// check registry
+	env.requireOffsetInRegistry(testlogName, len(testlines))
+
+	fi, err := os.Stat(env.abspath(testlogName))
+	if err != nil {
+		t.Fatalf("cannot stat file: %+v", err)
+	}
+
+	env.mustRemoveFile(testlogName)
+
+	// the second log line will not be picked up as scan_interval is set to one day.
+	env.mustWriteLinesToFile(testlogName, []byte("first line\nsecond log line\n"))
+
+	// new two events arrived
+	env.waitUntilEventCount(1)
+
+	cancelInput()
+	env.waitUntilInputStops()
+
+	identifier, _ := newINodeDeviceIdentifier(nil)
+	src := identifier.GetSource(loginp.FSEvent{Info: fi, Op: loginp.OpCreate, NewPath: env.abspath(testlogName)})
+	env.requireOffsetInRegistryByID(src.Name(), len(testlines))
```

Comment:
shall we validate the state of the new file as well? To ensure that both files are known correctly to the registry?

Reply:
The new file is not picked up because `prospector.scanner.check_interval` is set to a day.

Reply:
I see. This means that the test is very timing sensitive. Given that we check for the presence of a single event that already exists, the test can't really tell if it was successful because the execution was correct, or if it was successful because of some bad timing. Plus we do write the lines to a new file, not the old one, which still won't tell us if the harvester was correctly shutdown.

Maybe a more thorough validation step would be to 1. remove the file, 2. wait for the number of go-routines being reduced (harvester is gone) 3. append another line to the removed file. The third step will only work on *nix (ignore the step on windows). For this to work you need to keep the file open for the duration of the test.